adafruit_requests.py

```diff
@@ -157,7 +239,7 @@ def _recv_into(self, buf: bytearray, size: int = 0) -> int:
             read_size = len(b)
             buf[:read_size] = b
             return read_size
-        return self.socket.recv_into(buf, size)
+        return cast("SupportsRecvInto", self.socket).recv_into(buf, size)
```

Comment:
This `cast()` is needed for mypy to understand that we can safely call `recv_into()` on `self.socket` here because the legacy type of CircuitPython socket (represented by the `LegacyCircuitPythonSocketType` and the `self._backwards_compatible` flag checked above being `True`) does not have a `recv_into()` method.

Reply:
@kevincon it seems that the change here caused the library to stop working on microcontrollers with this error:
```
code.py output:
Connecting to AP...
Connected to WBurn-2G 	RSSI: -50
Fetching text from http://wifitest.adafruit.com/testwifi/index.html
Traceback (most recent call last):
  File "code.py", line 58, in <module>
  File "adafruit_requests.py", line 847, in get
  File "adafruit_requests.py", line 719, in request
  File "adafruit_requests.py", line 210, in __init__
  File "adafruit_requests.py", line 297, in _readto
  File "adafruit_requests.py", line 241, in _recv_into
TypeError: function takes 2 positional arguments but 3 were given

Code done running.
```
I think we'll need to revert it for now. Curious if you have any ideas that could make it worth in both environments?

Reply:
Actually maybe this is not what is causing the error. My mistake. I reverted it in my local copy, but still seeing that error